### DISCUSSION TITLE
Steel vs others for Helix?

### DISCUSSION THREAD

**Discussion by emummel20** [mattwparas/steel#119] [General]
Hey I was wondering what the difference would be between Steel and the rust-scheme alternatives and why it was selected for Helix? Alternatives being eg. [oxischeme](https://fitzgen.github.io/oxischeme/oxischeme/index.html), [vonuvoli](https://github.com/volution/vonuvoli-scheme), [marwood](https://github.com/strtok/marwood/) ..

Note, I am _not_ complaining or wishing it were different. I like scheme and am actually quite happy with the plugin lang selection if we do go with Steel, but I'm just super curious why Steel.

**Comment by mattwparas** [comment]
So let me clarify a few things - I do not speak for any of the Helix maintainers, and I am not a member of their team.

Steel has not _yet_ been accepted as the language of choice, it is a just a good candidate. The reason we've gotten to this point is that I did the integration of Steel into Helix. There had been a lot of discussion on what scripting language to use, but nobody had actually gotten as far as doing the integration.

I started doing the integration to stress test the Steel implementation and integration API, and had good success. As a result, there is an open PR into Helix adding Steel, that has garnered some attention. I have not done a deep dive into those schemes - at a glance, it looks like oxischeme hasn't been updated in 8 years, and the other two also haven't been updated in about a year. I'm sure the latter two are interesting, but there was a bit of a first movers advantage that I personally wrote the integration of steel into helix, having also written steel myself, which made it easier for me than someone just using the steel APIs without much familiarity with the internals.

**Reply by emummel20** [reply]
Got it, thanks.

**Reply by emummel20** [reply]
@mattwparas one more quick question: I get fib, but what's the "ack" performance test?

**Reply by mattwparas** [reply]
```scheme
(define (ackermann m n)
  (cond [(equal? m 0) (+ n 1)]
        [(equal? n 0) (ackermann (- m 1) 1)]
        [else (ackermann (- m 1) (ackermann m (- n 1)))]))

```

Just tests some math and function calling
